plugins/org.eclipse.fordiac.ide.model.search/src/org/eclipse/fordiac/ide/model/search/types/GlobalConstantsTypeInstanceSearch.java

```diff
@@ -0,0 +1,173 @@
+import org.eclipse.fordiac.ide.model.search.ISearchSupport;
+import org.eclipse.fordiac.ide.model.search.LocalLiveSearchContext;
+import org.eclipse.fordiac.ide.model.typelibrary.GlobalConstantsEntry;
+
+public class GlobalConstantsTypeInstanceSearch extends IEC61499ElementSearch {
+
+	public GlobalConstantsTypeInstanceSearch(final GlobalConstantsEntry gcEntry) {
+		super(new LocalLiveSearchContext(gcEntry.getTypeLibrary()), createSearchFilter(gcEntry),
+				new GlobalConstantsTypeInstanceSearchChildrenProvider());
+	}
+
+	private static IEC61499SearchFilter createSearchFilter(final GlobalConstantsEntry gcEntry) {
+		final GlobalConstantsMatcher matcher = new GlobalConstantsMatcher(gcEntry);
+		return searchCandidate -> searchCandidate != null && switch (searchCandidate) {
+		case final VarDeclaration varDecl -> isGlobalConstantValue(varDecl, matcher);
+		case final Attribute attr -> isGlobalConstantValue(attr, matcher);
+		case final Algorithm alg -> isGlobalConstantValue(alg, matcher);
+		case final Method meth -> isGlobalConstantValue(meth, matcher);
+		case final STFunctionBody func -> isGlobalConstantValue(func, matcher);
+		default -> false;
+		};
+	}
+
+	private static boolean isGlobalConstantValue(final VarDeclaration varDecl, final GlobalConstantsMatcher matcher) {
+		if (varDecl.getValue() == null) {
+			return false;
+		}
+		final String value = varDecl.getValue().getValue();
+		if (value == null || value.isBlank() || value.chars().noneMatch(Character::isLetter)
+				|| VariableOperations.isSimpleInitialValue(varDecl)) {
+			return false;
+		}
+
+		return isGlobalConstantValue((EObject) varDecl, matcher);
+	}
+
+	private static boolean isGlobalConstantValue(final Attribute attr, final GlobalConstantsMatcher matcher) {
+		final String value = attr.getValue();
+		if (value == null || value.isBlank() || value.chars().noneMatch(Character::isLetter)
+				|| VariableOperations.isSimpleAttributeValue(attr)) {
+			return false;
+		}
+
+		return isGlobalConstantValue((EObject) attr, matcher);
+	}
+
+	private static boolean isGlobalConstantValue(final EObject context, final GlobalConstantsMatcher matcher) {
+		final ISearchSupport searchSupport;
+		switch (context) {
+		case final VarDeclaration varDecl: {
+			searchSupport = ISearchFactory.createSearchSupport(varDecl, VarDeclaration.class);
+			break;
+		}
+		case final Attribute attr: {
+			searchSupport = ISearchFactory.createSearchSupport(attr, Attribute.class);
+			break;
+		}
+		case final STAlgorithm alg: {
+			searchSupport = ISearchFactory.createSearchSupport(alg, STAlgorithm.class);
+			break;
+		}
+		case final STMethod meth: {
+			searchSupport = ISearchFactory.createSearchSupport(meth, STMethod.class);
+			break;
+		}
+		case final STFunctionBody body: {
+			searchSupport = ISearchFactory.createSearchSupport(body, STFunctionBody.class);
+			break;
+		}
+		default: {
+			return false;
+		}
+		}
+
+		return searchSupport.search(matcher).count() > 0;
```

Comment:
I would suggest to simplify this to:

```suggestion
		final ISearchSupport searchSupport = ISearchFactory.createSearchSupport(context,
				context.eClass().getInstanceClass());
		return searchSupport != null && searchSupport.search(matcher).count() > 0;
```